rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -361,11 +409,19 @@ async def get_graphs_meta(
     if user_id and filter_by != "template":
         where_clause["userId"] = user_id
 
-    graphs = await AgentGraph.prisma().find_many(
-        where=where_clause,
-        distinct=["id"],
-        order={"version": "desc"},
-    )
+    if include_executions:
+        graphs = await AgentGraph.prisma().find_many(
+            where=where_clause,
+            distinct=["id"],
+            order={"version": "desc"},
+            include=AgentGraphInclude(AgentGraphExecution={"include": {"AgentNodeExecutions": True}}),  # type: ignore
+        )
+    else:
+        graphs = await AgentGraph.prisma().find_many(
+            where=where_clause,
+            distinct=["id"],
+            order={"version": "desc"},
+        )
```

Comment:
Unnecessary duplication:
```suggestion
    graphs = await AgentGraph.prisma().find_many(
        where=where_clause,
        distinct=["id"],
        order={"version": "desc"},
        include=AgentGraphInclude(
            AgentGraphExecution={"include": {"AgentNodeExecutions": True}}
        )
        if include_executions
        else None,
    )
```

And `# type: ignore` which really isn't necessary.

Reply:
# type: ignore is cause the type system can't recognise a valid prisma query...

Reply:
yes, it can:
![image](https://github.com/user-attachments/assets/658082c9-5acb-47d5-a096-3b6beeaee8b8)